core/src/main/java/org/bitcoinj/core/Block.java

```diff
@@ -256,11 +280,15 @@ protected void parse() throws ProtocolException {
         length = cursor - offset;
     }
 
-    public static Block createGenesis(NetworkParameters n) {
-        Block genesisBlock = new Block(n, BLOCK_VERSION_GENESIS);
+    public static Block createGenesis(NetworkParameters n, long time, long difficultyTarget, long nonce) {
+        Transaction t = createGenesisTransaction(n, genesisTxInputScriptBytes, FIFTY_COINS, genesisTxScriptPubKeyBytes);
+        return new Block(n, BLOCK_VERSION_GENESIS, time, difficultyTarget, nonce, Collections.singletonList(t));
```

Comment:
We could use the appropriate TARGET constant here, rather than passing the value in.

Reply:
We could do that, but there could conceivably be use cases (test cases) where we want the difficulty to be different.

The TARGET constants from the NetworkParameters subclasses are no accessible here, but that could be changed, of course.